crates/cxx-qt-build/src/lib.rs

```diff
@@ -478,7 +478,14 @@ impl CxxQtBuilder {
         let mut cc_builder_whole_archive_files_added = false;
         // Run moc on C++ headers with Q_OBJECT macro
         for qobject_header in self.qobject_headers {
-            let moc_products = qtbuild.moc(&qobject_header.path);
+            let moc_uri_opt = qobject_header
+                .qml_metadata
+                .iter()
+                .flat_map(|qml_metadata| ["-Muri=", &qml_metadata.uri, " "])
```

Comment:
is the `" "` here trying to make a space between multiple uris or something ? So that you result in `-Muri=abc -Muri=def` ?

If so I think it would be easier to read if you built them to a list then joined with a " " ?

eg something like this seems more readable, unless i'm misunderstanding what is happening ?

```rust
            let moc_uri_opt = qobject_header
                .qml_metadata
                .iter()
                .flat_map(|qml_metadata| ["-Muri=", &qml_metadata.uri])
                .collect::<Vec<&str>>()
                .join(" ");
```

Something like

Reply:
You are correct that this was the intent. Your suggestion looks nice but would produce a slightly different output `-Muri= abc -Muri= def`(note the spaces).

I chose to do something I should have done from the start and just pass all uris to the moc function so it can figure out how to create the arguments required for the moc compiler. this would make it much easier to use this in the future.

Reply:
There is still a small issue when uris is an empty iterator. I will make a fix for that at a later time.

Reply:
should be fixed ow